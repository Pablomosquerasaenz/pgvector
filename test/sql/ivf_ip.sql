SET enable_seqscan = off;

CREATE TABLE t (val vector(3));
INSERT INTO t (val) VALUES ('[0,0,0]'), ('[1,2,3]'), ('[1,1,1]'), ('[1,2,4]'), (NULL);
CREATE INDEX ON t USING ivf (val vector_ip_ops) WITH (lists = 1, quantizer = 'SQ8');

SELECT * FROM t ORDER BY val <#> '[3,3,3]';
SELECT COUNT(*) FROM (SELECT * FROM t ORDER BY val <#> (SELECT NULL::vector)) t2;

DROP TABLE t;
