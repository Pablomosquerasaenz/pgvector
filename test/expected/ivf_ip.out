SET enable_seqscan = off;
CREATE TABLE t (val vector(3));
INSERT INTO t (val) VALUES ('[0,0,0]'), ('[1,2,3]'), ('[1,1,1]'), ('[1,2,4]'), (NULL);
CREATE INDEX ON t USING ivf (val vector_ip_ops) WITH (lists = 1, quantizer = 'SQ8');
SELECT * FROM t ORDER BY val <#> '[3,3,3]';
   val   
---------
 [1,2,4]
 [1,2,3]
 [1,1,1]
 [0,0,0]
(4 rows)

SELECT COUNT(*) FROM (SELECT * FROM t ORDER BY val <#> (SELECT NULL::vector)) t2;
 count 
-------
     4
(1 row)

DROP TABLE t;
